compass/landice/tests/thwaites/mesh.py

```diff
@@ -0,0 +1,218 @@
+        # the Amundsen Sean Embayment, to speed up interpolation.
+        # This could also be replaced with the full Antarctic Ice Sheet
+        # dataset.
+        logger.info('calling interpolate_to_mpasli_grid.py')
+        args = ['interpolate_to_mpasli_grid.py', '-s',
+                'antarctica_1km_2020_10_20_ASE.nc', '-d',
+                'ase_1km_preCull.nc', '-m', 'b', '-t']
+
+        check_call(args, logger=logger)
+
+        # This step is only necessary if you wish to cull a certain
+        # distance from the ice margin, within the bounds defined by
+        # the GeoJSON file.
+        cullDistance = section.get('cull_distance')
+        if float(cullDistance) > 0.:
+            logger.info('calling define_cullMask.py')
+            args = ['define_cullMask.py', '-f',
+                    'ase_1km_preCull.nc', '-m'
+                    'distance', '-d', cullDistance]
+
+            check_call(args, logger=logger)
+        else:
+            logger.info('cullDistance <= 0 in config file. '
+                        'Will not cull by distance to margin. \n')
+
+        # This step is only necessary because the GeoJSON region
+        # is defined by lat-lon.
+        logger.info('calling set_lat_lon_fields_in_planar_grid.py')
+        args = ['set_lat_lon_fields_in_planar_grid.py', '-f',
+                'ase_1km_preCull.nc', '-p', 'ais-bedmap2']
+
+        check_call(args, logger=logger)
+
+        logger.info('calling MpasMaskCreator.x')
+        args = ['MpasMaskCreator.x', 'ase_1km_preCull.nc',
+                'thwaites_mask.nc', '-f', 'thwaites_minimal.geojson']
+
+        check_call(args, logger=logger)
+
+        logger.info('culling to geojson file')
+        dsMesh = xarray.open_dataset('ase_1km_preCull.nc')
+        thwaitesMask = xarray.open_dataset('thwaites_mask.nc')
+        dsMesh = cull(dsMesh, dsInverse=thwaitesMask, logger=logger)
+        write_netcdf(dsMesh, 'thwaites_culled.nc')
+
+        logger.info('Marking horns for culling')
+        args = ['mark_horns_for_culling.py', '-f', 'thwaites_culled.nc']
+        check_call(args, logger=logger)
+
+        logger.info('culling and converting')
+        dsMesh = xarray.open_dataset('thwaites_culled.nc')
+        dsMesh = cull(dsMesh, logger=logger)
+        dsMesh = convert(dsMesh, logger=logger)
+        write_netcdf(dsMesh, 'thwaites_dehorned.nc')
+
+        logger.info('calling create_landice_grid_from_generic_MPAS_grid.py')
+        args = ['create_landice_grid_from_generic_MPAS_grid.py', '-i',
+                'thwaites_dehorned.nc', '-o',
+                'Thwaites_1to8km.nc', '-l', levels, '-v', 'glimmer',
+                '--beta', '--thermal', '--obs', '--diri']
+
+        check_call(args, logger=logger)
+
+        logger.info('calling interpolate_to_mpasli_grid.py')
+        args = ['interpolate_to_mpasli_grid.py', '-s',
+                'antarctica_1km_2020_10_20_ASE.nc',
+                '-d', 'Thwaites_1to8km.nc', '-m', 'b', '-t']
```

Comment:
Speaking of potential speedup, switching the interpolation method from bilinear to delaunay (barycentric) might help here, but then I also see the `-t` option here that would only interpolate thickness.  Don't we want to interpolate everything in this step?

Reply:
I remember bilinear being _much_ faster than barycentric, unless we save the barycentric weights somewhere.

Reply:
We eventually want to interpolate everything, but we have not done that within COMPASS before, so I was thinking that should be in a separate PR. It will be a considerable effort, and will include switching over to the conservative remapping as well.

Reply:
Yes bilinear is faster for one or a few fields, but the barycentric only calculates weights once, so it may be faster when there are a lot of fields.

I see about not doing the final step in this PR.  What do you think about dropping `-t` anyway so what comes out of COMPASS is at least a rough draft of all the data off the shelf?  Then we can still rerun the interpolation manually.

Reply:
It seems like you might not want to have your own tool for this.  I think this is something that `ESMF_RegridWeightGen` and `ncremap` should be able to handle.  You can use `pyremap` to generate the MPAS script files, create the weight files, and do the remapping:
https://github.com/MPAS-Dev/pyremap/blob/master/examples/make_mpas_to_Antarctic_stereo_mapping.py
It makes calls to `E3SM_RegridWeightGen` and `ncremap`.

Reply:
> Yes bilinear is faster for one or a few fields, but the barycentric only calculates weights once, so it may be faster when there are a lot of fields.

@matthewhoffman I confirmed that barycentric takes ~25% longer than bilinear. That's not a huge deal for a smallish mesh like this, but I'd push to leave it as-is for this PR and then update interpolation for all real-geometry test-cases (i.e., Humboldt, Thwaites, AIS) in another PR following @xylar's  suggestion.

Reply:
Thanks for testing that out.  I agree bilinear vs. barycentric doesn't matter for this case and the performance differences are too small to matter.

In the long run Xylar's suggestion of pyremap sounds wise.  Our current tool is highly specialized for our workflow, however, so we'll wait until we reach a point of specifically needing new capabilities before looking into replacing it.